src/src/com/microsoft/aad/adal/HttpWebRequest.java

```diff
@@ -109,8 +109,6 @@ private HttpURLConnection setupConnection() throws IOException {
             connection.setRequestProperty(header, mRequestHeaders.get(header));
         }
 
-        // Avoid reuse of existing sockets to avoid random EOF errors
-        System.setProperty("http.keepAlive", "false");
```

Comment:
I would recommend that this change be put in selfhost and tested before releasing to the public.